test-manager/test_macro/src/lib.rs

```diff
@@ -76,52 +76,79 @@ fn create_test(
+                    Box::pin(async move {
+                        let result = #func_name(rpc, *mullvad_client.clone()).await;
+                        // If default settings are not retrieved, retrieve them and cleanup
+                        match crate::tests::get_default_settings() {
+                            Some(default_settings) => {
+                                let default_settings = Some(default_settings);
+                                let mullvad_client = Some(*mullvad_client);
+                                if #should_cleanup {
+                                    crate::tests::cleanup_after_test(default_settings, mullvad_client).await?;
+                                }
+                            }
+                            None => {
+                                let default_settings = Some(crate::tests::set_default_settings(&mut mullvad_client).await.unwrap());
+                                let mullvad_client = Some(*mullvad_client);
+                                if #should_cleanup {
+                                    crate::tests::cleanup_after_test(default_settings, mullvad_client).await?;
+                                }
+                            }
+                        }
+                        result
+                    })
+                }
             }
-        }
-    } else {
-        quote! {
-            |rpc: test_rpc::ServiceClient,
-            mullvad_client: Box<dyn std::any::Any + Send>,| {
-                Box::pin(async move {
-                    let result = #func_name(rpc).await;
-                    // TODO: We should make a third crate for types to avoid this circular dependency
-                    // and hardcoding names of types
-                    // Cleanup with the newest management service and if that's not available then skip
-                    // cleanup
-                    if let Ok(mullvad_client) = mullvad_client.downcast::<ManagementServiceClient>() {
-                        #cleanup
-                    }
-                    result
-                })
+        },
+        MullvadClient::Old { mullvad_client_type, .. } => {
+            let mullvad_client_type = *mullvad_client_type;
+            quote! {
+                |rpc: test_rpc::ServiceClient,
+                mullvad_client: Box<dyn std::any::Any + Send>,|
+                {
+                    use std::any::Any;
+                    let mullvad_client = mullvad_client.downcast::<#mullvad_client_type>().expect("invalid mullvad client");
+                    Box::pin(async move {
+                        let result = #func_name(rpc, *mullvad_client.clone()).await;
+                        // Do not set the default settings since we only have access to the old
+                        // management interface
+                        if #should_cleanup {
+                            crate::tests::cleanup_after_test(crate::tests::get_default_settings(), None).await?;
```

Comment:
This is always a no-op now, right? If so, I think it should be removed.

Reply:
This is not necessarily a no-op since it will run whatever cleanup we want that is independent of settings. That's nothing currently which perhaps makes it pointless. I'll remove it.